speech_recognition/__init__.py

```diff
@@ -1390,7 +1390,25 @@ def recognize_tensorflow(self, audio_data, tensor_graph='tensorflow-data/conv_ac
             for node_id in top_k:
                 human_string = self.tflabels[node_id]
                 return human_string
-
+            
+    def recognize_vosk(self, audio_data, language='en'):
+        from vosk import Model, KaldiRecognizer
+
+        if not os.path.exists("model"):
+            return "Please download the model from https://github.com/alphacep/vosk-api/blob/master/doc/models.md and unpack as 'model' in the current folder."
+            exit (1)
+        
+        assert isinstance(audio_data, AudioData), "Data must be audio data"
+        
+        if not hasattr(self, 'vosk_model'):
```

Comment:
os.path.exists should also go here since you create the model and before that you need to check the path. There is no need to check the path on every request:
````
        if not hasattr(self, 'vosk_model'):
            if not os.path.exists("model"):
                return "Please download the model from https://github.com/alphacep/vosk-api/blob/master/doc/models.md and unpack as 'model' in the current folder."
                exit (1)
            self.vosk_model = Model()
```

Ideally you'd have a model path instead of fixed value "model".  Like here:

https://github.com/Uberi/speech_recognition/blob/c89856088ad81d81d38be314e3db50905481c5fe/speech_recognition/__init__.py#L782

Reply:
Done.
You helped so much.
Should i rewrite ownership of my fork to you????

Reply:
No need in that, better make a proper pull which is working.